src/main/java/com/puppycrawl/tools/checkstyle/checks/OrderedPropertiesCheck.java

```diff
@@ -0,0 +1,189 @@
+
+                final int lineNumber = getLineNumber(fileText, propKey);
+                // Number of occurrences is number of duplications + 1
+                log(lineNumber, MSG_KEY, propKey, previousProp);
+            }
+
+            previousProp = propKey;
+        }
+    }
+
+    /**
+     * Method returns line number the key is detected in the checked properties
+     * files first.
+     *
+     * @param fileText
+     *            {@link FileText} object contains the lines to process
+     * @param keyName
+     *            key name to look for
+     * @return line number of first occurrence. If no key found in properties
+     *         file, 1 is returned
+     */
+    private static int getLineNumber(FileText fileText, String keyName) {
+        final Pattern keyPattern = getKeyPattern(keyName);
+        int lineNumber = 1;
+        final Matcher matcher = keyPattern.matcher("");
+        for (int index = 0; index < fileText.size(); index++) {
+            final String line = fileText.get(index);
+            matcher.reset(line);
+            if (matcher.matches()) {
+                break;
+            }
+            ++lineNumber;
+        }
+        // -1 as check seeks for the first duplicate occurrence in file,
+        // so it cannot be the last line.
+        if (lineNumber > fileText.size() - 1) {
+            lineNumber = 0;
+        }
+        return lineNumber;
+    }
+
+    /**
+     * Method returns regular expression pattern given key name.
+     *
+     * @param keyName key name to look for
+     * @return regular expression pattern given key name
+     */
+    private static Pattern getKeyPattern(String keyName) {
+        final String keyPatternString = "^" + SPACE_PATTERN.matcher(keyName)
+                .replaceAll(Matcher.quoteReplacement("\\\\ ")) + "[\\s:=].*$";
+        return Pattern.compile(keyPatternString);
+    }
+
+    /**
+     * Private property implementation that keeps order of properties like in file.
+     *
+     * @noinspection ClassExtendsConcreteCollection, SerializableHasSerializationMethods
+     */
+    private static class SequencedProperties extends Properties {
+
+        private static final long serialVersionUID = 1L;
+
+        /**
+         * Holding the keys in the same order than in the file.
+         */
+        private final List<Object> keyList = new ArrayList();
+
+        /**
+         * Returns a copy of the keys.
+         */
+        @Override
+        public Enumeration<Object> keys() {
+            return Collections.enumeration(keyList);
+        }
+
+        /**
+         * Puts the value into list by its key.
+         * @noinspection UseOfPropertiesAsHashtable
+         */
+        @Override
+        public Object put(Object key, Object value) {
+            if (!containsKey(key)) {
+                keyList.add(key);
+            }
+
+            return super.put(key, value);
+        }
+
+        @Override
+        public boolean equals(Object o) {
```

Comment:
Why are equals and hashcode needed? If there is a violation we can just suppress it like `UniquePropertiesCheck` does.

Reply:
No, I do not need equals and hashCode. I was not able to suppress it. I have got something, that suppression is not allowed...
I struggling with the strict rules. It is my first contribution. Please give me some advice here.

Reply:
Which CI gave you the violation that you needed equals/hashcode? Suppression is dependent on who is requiring it.

Reply:
it was spotbugs, EQ_DOESNT_OVERRIDE_EQUALS
I was able to solve it by editing spotbugs-exclude.xml